R/soar-Dashboard.R

```diff
@@ -5,6 +5,54 @@ library(leaflet)
 library(rgbif)
 library(shinycssloaders)
 
+min_choice <- function() {
```

Comment:
Also it is not a good practice to create functions that have not arguments and only call global variables. This makes understanding the scope and flow of the function more difficult. Also it would be helpful if you annotated the code so that it was clear what the purpose of the function is. It appears to be just trying to come up with the column indices for the fields that are desired. Is that correct? There is a much easier way to do this
```r
# assume the object dat is a data.frame you wish to only take certain columns of 
true = names(dat) %in% columns_of_interest
dat_sub = dat[ , true]

# a better approach that is easier for a human to read is: 
dat_sub = subset(dat, select = columns_of_interest)

Reply:
I wrote the method like this because it returned the list in a format that the checkboxGroupInput() understood as a valid 'choices' variable. It has to be inputted as a list with "option name" = ID#. So it's not really a matter of getting what columns are true as the names and numbers and then formatting them correctly. I have also tried the method you're suggesting a few times and I just can't seem to get it to work properly. I had to make these functions separate from the rest of the code because the choices variable would not allow me to put the code there and everywhere else put it caused an error except there. Is there a designated spot for methods in the ui? I can, however, combine the min_choice() and def_ choice functions, I'll get started on that.

Reply:
Sorry I'm still having trouble understanding what the goal of the function is. Is it to create an empty list with the correct number of elements and each element to have the correct name? Also don't forget to merge the `cus_choice` function as well.